src/Microsoft.VisualStudio.ProjectSystem.Managed/Tree/Dependencies/Subscriptions/RuleHandlers/DependenciesRuleHandlerBase.cs

```diff
@@ -109,10 +109,10 @@ void HandleChangesForRule(bool resolved, IProjectChangeDescription projectChange
 
                 IDependencyModel CreateDependencyModelForRule(string itemSpec, IProjectRuleSnapshot projectRuleSnapshot)
                 {
-                    IImmutableDictionary<string, string> properties = projectRuleSnapshot.GetProjectItemProperties(itemSpec);
+                    IImmutableDictionary<string, string> properties = projectRuleSnapshot.GetProjectItemProperties(itemSpec)!;
 
                     string originalItemSpec = resolved
-                        ? properties.GetStringProperty(ResolvedAssemblyReference.OriginalItemSpecProperty)
+                        ? properties.GetStringProperty(ResolvedAssemblyReference.OriginalItemSpecProperty) ?? itemSpec
```

Comment:
Behaviour change. if somehow the original item spec property was null, use the item spec.

Reply:
We should probably change these methods to return empty to behave identiical to MSBuild; if a property doesn't exist its "empty". Probably need to rationalize these extension methods with this `ProjectRuleSnapshotExtensions`.